File: 0053-maximum-subarray/0053-maximum-subarray.c
int maxSubArray(int* nums, int numsSize){
        int max_sum = nums[0];
        int current_sum = nums[0];

        for(int i = 1; i < numsSize; i++){
            current_sum = fmax(nums[i], current_sum + nums[i]);
            max_sum = fmax(max_sum, current_sum);
        }

        return max_sum;
}

File: 0053-maximum-subarray/0053-maximum-subarray.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut max_sum = nums[0];
        let mut current_sum = nums[0];

        for &num in nums.iter().skip(1) {
            current_sum = num.max(current_sum + num);
            max_sum = max_sum.max(current_sum);
        }

        max_sum
    }
}
